src/ast.alp

```diff
@@ -22,4 +59,32 @@ let float line val = Float {line=line, val=val}
 
 let float_val Float {line=_, val=v} = v
 
-let string line val = String {line=line, val=val}
\ No newline at end of file
+let string line val = String {line=line, val=val}
+
+let bind name line bound =
+  Binding {  line=line
+           , name=name
+           , bound=bound
+           , body=None
+           , typ=None
+          }
+
+let bind name line bound body =
+  Binding {  line=line
+           , name=name
+           , bound=bound
+           , body=Some body
+           , typ=None
+          }
+
+let bind_body Binding b_rec body = Binding {body=Some body | b_rec}
```

Comment:
Although I know the parse rules, I still find this confusing when quickly scanning through the code. I either intepret it as:
* a function taking three arguments (Binding, b_rec and body)
* or a function taking one argument, and pattern matching against a two argument constructor.

Only the knowledge that there is no such thing as a two argument constructor, and that Binding takes a type argument, disambiguate this so I can parse it as what it actually means, namely: 
`let bind_body (Binding b_rec) body ...`

I wonder if we shouldn't try to encourage this less ambiguous style.